lib/logstash/outputs/elasticsearch/license_check.rb

```diff
@@ -0,0 +1,29 @@
+module LogStash
+  module Outputs
+    module ElasticSearchPoolMixin
+      module LicenseChecker
+
+        # This mixin is used to externalize the license checking behaviour of the LogStash::Outputs::ElasticSearch::HttpClient::Pool class.
+
+        # Perform a license check
+        # @param url [LogStash::Util::SafeURI]
+        # @param meta [Hash]
+        def license_check!(url, meta)
+          if oss? || valid_es_license?(url)
```

Comment:
yeah,  I considered this, and settled for only exposing just a *main* `license_check!` method because I do not think more will be required on the data_streams side to implement the check but I think `oss?` and `valid_es_license?` should probably go there too. I also wanted to avoid performing too deep a refactoring just for a slight change in the check logic. This struck me as an acceptable way to externalize the check - I will definitely add a *big* comment about its usage and context; the fact it is only meant to live within the Pool class. 
I know this a rather loose integration and I'll see how I can add relevant specs to make sure the *api* for this mixin is somehow protected.

Reply:
So I opted to also extract `oss?`, `valid_es_license?`, `log_license_deprecation_warn` together with the main entry point ` license_check! `.